cou/steps/analyze.py

```diff
@@ -111,4 +143,9 @@ def current_cloud_os_release(self) -> Optional[OpenStackRelease]:
         :return: OpenStack release codename
         :rtype: OpenStackRelease
         """
-        return min((app.current_os_release for app in self.apps), default=None)
+        control_plane = (app for app in self.apps_control_plane)
+        all_apps = chain(control_plane, self.apps_data_plane)
```

Comment:
I agree that Gabriel solution is nicer, I did not understand purpose of `control_plane = (app for app in self.apps_control_plane)`, what is different than using `self.apps_control_plane`?

Since both apps_control_plane and apps_data_plane are defined as `Iterable[OpenStackApplication]`, can we use set instead of list?

Reply:
using set for `apps_control_plane` and `apps_data_plane` can mess with the upgrade order. I think they should stay as list

Reply:
Generator and chaining is more effective since it does not create new data structure. But I am ok with any other solution.

Reply:
Why do we care about the order here? Don't we just need to find a minimal openstack release out of `all_apps`?